_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,363 @@
+  entityOperator:
+    topicOperator: {}
+    userOperator: {}
+```
+
+Once the cluster is running, lets deploy some topics where we will send and receive the messages.
+Here is an example topic configuration. You can change it to suit your requirements.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaTopic
+metadata:
+  name: my-topic-1
+  labels:
+    strimzi.io/cluster: <CLUSTER-NAME>
+spec:
+  partitions: 10
+  replicas: 3
+  config:
+    retention.ms: 7200000
+    segment.bytes: 1073741824
+    # ...
+```
+
+Now we require a kafka user.
+We will configure the kafka user with ACL rules that specify permission to produce and consume topics from the Kafka brokers.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaUser
+metadata:
+  name: <KAFKA-USER>
+  labels:
+    strimzi.io/cluster: <CLUSTER-NAME>
+spec:
+  authentication:
+    type: tls
+  authorization:
+    type: simple
+    acls:
+      - resource:
+          type: topic
+          name: my-topic-1
+          patternType: literal
+        operation: Write
+        host: "*"
+      - resource:
+          type: topic
+          name: my-topic-1
+          patternType: literal
+        operation: Read
+        host: "*"
+      - resource:
+          type: topic
+          name: my-topic-1
+          patternType: literal
+        operation: Describe
+        host: "*"   
+      - resource:
+           type: topic
+           name: my-topic-1
+           patternType: literal
+        operation: DescribeConfigs
+        host: "*"
+      - resource:
+           type: topic
+           name: my-topic-1
+           patternType: literal
+        operation: AlterConfigs
+        host: "*"        
+      - resource:
+           type: cluster
+           name: <CLUSTER-NAME>
+           patternType: literal
+        operation: Alter
+        host: "*"
+      - resource:
+           type: cluster
+           name: <CLUSTER-NAME>
+           patternType: literal
+        operation: AlterConfigs
+        host: "*"        
+  # ...
+```
+
+### Creating a proposal `reassignment.json` file
+
+When you have a Kafka cluster running with brokers and topics, you are ready to create the proposal JSON file.
+
+Let us create a separate interactive pod.
+This interactive pod is used to run all the reassignment commands. 
+One question might bug you. What is the need of a separate pod? Can't we just use one of the broker pods?
+In answer to this question, running commands from within a broker is not good practice as it's insecure
+It will start another JVM inside the container designated for the broker and can cause disruption, cause the container to run out of memory and so on.
+So it is always better to avoid running the command from a broker pod.
+
+So now it's time to get our interactive pod up and running. You can use the following command:
+
+```sh
+kubectl run --restart=Never --image=quay.io/strimzi/kafka:0.27.0-kafka-3.0.0 <INTERACTIVE-POD-NAME> -- /bin/sh -c "sleep 3600"
```

Comment:
Maybe mention that the `sleep` means we have 1 hour to use this pod? That way someone following along won't be surprised if their pod stops working.